R/cohenFleissKappa.R

```diff
@@ -0,0 +1,202 @@
+    100 * options[["kappaConfidenceIntervalValue"]],
+    digits = 3,
+    drop0trailing = TRUE
+  )
+  
+  
+  if(ready){
+    #Create all pairs
+    variables <- colnames(dataset)
+    allPairs <- .getPairs(variables)
+    allDataframes <- .getDataframes(dataset, allPairs)
+    allPairStrings <- paste(allPairs$x, allPairs$y, sep = " - ")
+    
+    #calculate Cohen's Kappas
+    allKappaData <- lapply(allDataframes, psych::cohen.kappa, alpha = 1 - options[["kappaConfidenceIntervalValue"]])
+    
+    #Extract Kappas and CIs
+    allKappas <- c()
+    allLowerBounds <- c()
+    allUpperBounds <- c()
+    
+    for(i in allKappaData){
+      kappaData <- i$confid
+      if(weighted){
+        k <- 2
+      }else{
+        k <-1
+      }
+      allKappas <- c(allKappas, kappaData[k,2])
+      allLowerBounds <- c(allLowerBounds, kappaData[k,1])
+      allUpperBounds <- c(allUpperBounds, kappaData[k,3])
+    }
+    
+    tableData <- list("ratings" = allPairStrings,
+                      "cKappa" = allKappas)
+    
+    if (options[["kappaIntervalOn"]]) {
+      jaspTable$addColumnInfo(name = "CIL", title = gettext("Lower"), type = "number", overtitle = gettextf("%s%% CI for Kappa", formattedCIPercent))
+      jaspTable$addColumnInfo(name = "CIU", title = gettext("Upper"), type = "number", overtitle = gettextf("%s%% CI for Kappa", formattedCIPercent))
+      tableData[["CIL"]] <- allLowerBounds
+      tableData[["CIU"]] <- allUpperBounds
+    }
+    
+    
+    jaspTable$setData(tableData)
+    jaspTable$addFootnote(gettextf('%i subjects/items and %i judges/measurements.', nrow(dataset), ncol(dataset)))
+    
+    #if weighted kappa option is on but data only has 2 levels
+    if(weighted && length(levels(unlist(dataset))) < 3)
+      jaspTable$addFootnote(gettext('Weighted Kappa is only meaningful for more than 2 levels. Calculated weighted Kappa is the same as unweighted Kappa.'))
+    
+  }
+  
+  return(jaspTable)
+}
+
+.fleissKappa <- function(dataset, options, ready){
+  
+  # Create the JASP Table
+  jaspTable <- createJaspTable(title = gettextf("Fleiss' Kappa"))
+  jaspTable$addColumnInfo(name = "ratings", title = gettext("Ratings"), type = "string")
+  jaspTable$addColumnInfo(name = "fKappa", title = gettext("Fleiss' Kappa"), type = "number")
+  jaspTable$position <- 2
+  
+  #dependencies
+  jaspTable$dependOn(
+    options = c(
+      "variables",
+      "fleissKappa",
+      "kappaIntervalOn",
+      "kappaConfidenceIntervalValue"
+    )
+  )
+
+  
+  formattedCIPercent <- format(
+    100 * options[["kappaConfidenceIntervalValue"]],
+    digits = 3,
+    drop0trailing = TRUE
+  )
+  
+  
+  # if (options[["kappaIntervalOn"]]) {
+  #   jaspTable$addColumnInfo(name = "CIL", title = gettext("Lower"), type = "number", overtitle = gettextf("%s%% CI for Kappa", formattedCIPercent))
+  #   jaspTable$addColumnInfo(name = "CIU", title = gettext("Upper"), type = "number", overtitle = gettextf("%s%% CI for Kappa", formattedCIPercent))
+  # }
```

Comment:
Yes, I have also looked into this. I will implement the CI when I have time. I will probably have to write a function 'from scratch' for this, because the irr package does not provide the SEs.